src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -335,7 +355,14 @@ protected void OnDocumentContextUpdated(DocumentId documentId)
 
             if (container != null)
             {
-                OnDocumentContextUpdated(documentId, container);
+                if (_isProjectUnloading)
```

Comment:
reading of mutable state outside of the lock.  How do we know this is safe? Do we have threading asserts about when these things will happen?